src/dashboard/src/contrib/mcp/client.py

```diff
@@ -47,6 +49,24 @@ def execute(self, uuid, choice, uid=None):
         gm_client.shutdown()
         return
 
+    def execute_unit(self, unit_id, choice, mscl_id=None, uid=None):
+        """Execute the jobs awaiting for approval associated to a given unit.
+
+        Use ``mscl_id`` to pass the ID of the chain link to restrict the
+        execution to a single microservice.
+        """
+        kwargs = {
+            'currentstep': Job.STATUS_AWAITING_DECISION,
+            'sipuuid': unit_id,
+        }
+        if mscl_id is not None:
+            kwargs['microservicechainlink_id'] = mscl_id
+        jobs = Job.objects.filter(**kwargs)
+        if len(jobs) < 1:
+            raise Exception('No job was found.')
```

Comment:
We can define this as a custom exception, `MCPNoJobFoundException` maybe? This would mean we can be more explicit about the exception we want to handle in calling functions. This reduces the risk of hiding other exceptions/bugs that might be in calling functions.

Reply:
I can't see the original message in this thread but I remember the gist was that we want to return something simple that the user can understand. I had imagined that you could make the exception specific, because at the end of the day, it is, and then from the dashboard still return a generic message to the user, with a specific message to the logger. I wonder would that work? Or is it likely not robust enough for what a user might experience in the UI?

Reply:
If I understand correctly what you mean, yes it would work, you can definitely create as many custom exceptions as you want and raise them and still result in the same behaviour where I have a catch-all up in the stack as I'm doing.

However I don't know if I see the benefit, to me it's like a `if False: ...` ([we used to have some](https://github.com/artefactual/archivematica-history/blob/407bcb082f382dcfc54c5ec9b22a819ee6903402/src/MCPServer/lib/archivematicaMCP.py#L241)) which I'd remove it right away because it's dead code. Wouldn't this be similar? Excepting that it's harder to reason about because you also have to check its usage outside the function?

In any case, I can see the value of making this available for a potential use case in the future. Let me know if this would work for you:

```python

# In contrib.mcp.client

class NoJobFoundError(Exception):
    def __init__(self, *args, **kwargs):
        try:
            message = args[0]
        except IndexError:
            message = "No job was found"
        super(MyError, self).__init__(message)


# Usage:
raise NoJobFoundError()
raise NoJobFoundError("custom error message")
```

Reply:
This looks good @sevein :)